### PR TITLE
Integrate AutoRound v0.3

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

feature

## Description
```python
modify:
        n_samples -> nsamples
        n_blocks -> nblocks
quant config add:
        act_bits: int = 32,
        act_group_size: int = None,
        act_sym: bool = None,
        act_dynamic: bool = True,

other config:
        multimodal:bool = False,
        use_layer_wise: bool = False, ->  low_cpu_mem_usage: bool = False, (AutoRound repo)
```
## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: neural_compressor/torch/algorithms/weight_only/autoround.py [hunk 1]
```diff
@@ -31,69 +31,95 @@ class AutoRoundQuantizer(Quantizer):
...
         Args:
-        quant_config (dict): Configuration for weight quantization (default is None).
-        quant_config={
-                    'layer1':##layer_name
-                    {
-                        'data_type': 'int',
-                        'bits': 4,
-                        'group_size': 32,
-                        'sym': False,
+            quant_config (dict): Configuration for weight quantization (default is None).
```
Comment (WeiweiZhang1) [inline neural_compressor/torch/algorithms/weight_only/autoround.py:67]: for layer quantization

File: neural_compressor/torch/algorithms/weight_only/autoround.py [hunk 2]
```diff
@@ -31,69 +31,95 @@ class AutoRoundQuantizer(Quantizer):
...
+                        'layer1':##layer_name
+                        {
+                            'data_type': 'int',
+                            'bits': 4,
+                            'group_size': 32,
+                            'sym': False,
+                            'act_data_type': None,
+                            'act_bits': 32,
+                            'group_size': None,
+                            'sym': None,
```
Comment (WeiweiZhang1) [inline neural_compressor/torch/algorithms/weight_only/autoround.py:78]: repeat line

File: neural_compressor/torch/algorithms/weight_only/autoround.py [hunk 3]
```diff
@@ -150,23 +182,29 @@ def convert(self, model: torch.nn.Module, *args, **kwargs):
...
+            low_cpu_mem_usage=self.low_cpu_mem_usage,
         )
         model, weight_config = rounder.quantize()
         model.autoround_config = weight_config
         model = pack_model(model, weight_config, device=self.device, inplace=True)
         return model
 
 
-def get_dataloader(tokenizer, seqlen, dataset_name="NeelNanda/pile-10k", seed=42, bs=8, n_samples=512):
+def get_dataloader(tokenizer, seqlen, dataset_name="NeelNanda/pile-10k", seed=42, bs=8, nsamples=512):
```
Comment (WeiweiZhang1) [inline neural_compressor/torch/algorithms/weight_only/autoround.py:207]: nsamples=128

**Review by WeiweiZhang1 — APPROVED** [review]

File: neural_compressor/torch/quantization/config.py [hunk 4]
```diff
@@ -741,15 +745,16 @@ def __init__(
         low_gpu_mem_usage: bool = True,
         iters: int = 200,
         seqlen: int = 2048,
-        n_samples: int = 512,
+        nsamples: int = 512,
```
Comment (WeiweiZhang1) [inline neural_compressor/torch/quantization/config.py:748]: nsamples: int = 128,

File: neural_compressor/torch/algorithms/weight_only/autoround.py (hunk 3 above)
Comment (Kaihui-intel) [reply neural_compressor/torch/algorithms/weight_only/autoround.py]: https://github.com/intel/neural-compressor/pull/1925/commits/1c2e0d032009f49667e9c3e096f1d7cd921deb7f

File: neural_compressor/torch/quantization/config.py (hunk 4 above)
Comment (Kaihui-intel) [reply neural_compressor/torch/quantization/config.py]: https://github.com/intel/neural-compressor/pull/1925/commits/1c2e0d032009f49667e9c3e096f1d7cd921deb7f

File: neural_compressor/torch/algorithms/weight_only/autoround.py (hunk 2 above)
Comment (Kaihui-intel) [reply neural_compressor/torch/algorithms/weight_only/autoround.py]: fixed https://github.com/intel/neural-compressor/pull/1925/commits/6687d0fbf96b5b125e86d38fc4704c3603d53016

**Comment by thuang6** [comment]
It is v0.2.1 instead of v2.1

**Review by chensuyue — APPROVED** [review]

**Pull request merged**